File: src/com/leozhi/topic1370/Solution03.java
package com.leozhi.topic1370;

/**
 * @author leozhi
 */
public class Solution03 {
    public String sortString(String s) {
        int[] bucket = new int[26];
        for (char c : s.toCharArray()) {
            bucket[c - 'a']++;
        }
        StringBuilder res = new StringBuilder();
        while (res.length() < s.length()) {
            for (int i = 0; i < 26; i++) {
                if (bucket[i] > 0) {
                    res.append((char)(i + 'a'));
                    bucket[i]--;
                }
            }
            for (int i = 25; i >= 0; i--) {
                if (bucket[i] > 0) {
                    res.append((char)(i + 'a'));
                    bucket[i]--;
                }
            }
        }
        return res.toString();
    }
}

File: src/com/leozhi/topic1678/Solution03.kt
package com.leozhi.topic1678

/**
 * @author leozhi
 * 模拟
 * 通过
 * 164ms
 */
class Solution03 {
    fun interpret(command: String): String {
        val len = command.length
        var index = 0
        val stringBuilder = StringBuilder()
        while (index < len) {
            if (command[index] == 'G') {
                stringBuilder.append("G")
                ++index
            } else if (command[index] == '(' && command[index + 1] == ')') {
                stringBuilder.append("o")
                index += 2
            } else {
                stringBuilder.append("al")
                index += 4
            }
        }
        return stringBuilder.toString()
    }
}